event_mail/models/event.py

```diff
@@ -33,14 +30,7 @@ def _onchange_event_mail_template_id(self):
                 }))
             self.event_mail_ids = vals
 
-    @api.model
-    def _default_event_mail_ids(self):
-        if self.env.context.get('by_pass_config_template', False):
-            return super(EventEvent, self)._default_event_mail_ids()
-
-        event_mail_template_id = self.env['ir.values'].get_default(
-            'event.config.settings', 'event_mail_template_id')
-        if event_mail_template_id:
-            return super(EventEvent, self)._default_event_mail_ids()
-        else:
-            return []
+    @api.onchange('event_type_id')
+    def _onchange_type(self):
+        if self.event_type_id and self.event_mail_template_id:
+            self._onchange_event_mail_template_id()
```

Comment:
Why should we reset templates on event type change?

Reply:
This was bad implemented indeed. It's an onchange from the event module. New implementation coming so we just override it if a scheduling template is set